src/Controller/AdminController.php

```diff
@@ -920,6 +942,7 @@ public function tagsAction(Request $request, TagManager $tagManager)
      */
     public function pricingRuleSetsAction()
     {
+        $this->denyAccessUnlessGranted('ROLE_ADMIN');
```

Comment:
Are we sure it's better to add `denyAccessUnlessGranted`, than using `access_control` in `security.yml` (like you did for `/admin/settings`)? I see a lot of repetition here. 

Like written in the [docs](https://symfony.com/doc/5.4/security.html#add-code-to-deny-access), those are the only 2 ways. Maybe our URL structure does not fit this?

> You can define as many URL patterns as you need - each is a regular expression. BUT, only one will be matched per request: Symfony starts at the top of the list and stops when it finds the first match

Reply:
The job of the controller is to :

1. Validate user input
2. Check if the user is allowed to perform the action
3. Do business shenanigans (transform input data for the model or view)
4. Build a response if needed

To me, like network firewalls, the Symfony firewall is a nice new layer in the security onion – but does not dispense with security checks at the lowest level of entry points